backend/cfg/merge_straightline_blocks.ml

```diff
@@ -77,6 +77,8 @@ let rec merge_blocks (removed : Label.Set.t)
             b1_block.terminator <- b2_block.terminator;
             b1_block.exn <- b2_block.exn;
             b1_block.can_raise <- b2_block.can_raise;
+            Cfg_with_layout.replace_block_section cfg_with_layout b1_label
+              (Cfg_with_layout.get_section cfg_with_layout b2_label);
```

Comment:
My intuition is that if either b1 or b2 is cold, then the merged blocked should also be marked as cold. I think it is possible to reach L81 with b1 marked cold and b2 not marked cold, and end up with the merged blocked not marked cold when it should.

Reply:
Ah indeed, thanks. I tried a fix in the latest commit, what do you think about it?